shop/src/main/java/ro/msg/learning/shop/controller/ProductController.java

```diff
@@ -0,0 +1,71 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.dto.ProductAndCategoryDto;
+import ro.msg.learning.shop.exception.DuplicateResourceException;
+import ro.msg.learning.shop.exception.ResourceNotFoundException;
+import ro.msg.learning.shop.mapper.ProductMapper;
+import ro.msg.learning.shop.service.ProductService;
+
+import java.util.List;
+import java.util.UUID;
+
+@Validated
+@RequestMapping("/product")
+@RestController
+public class ProductController {
+    @Autowired
+    private ProductService productService;
+    @Autowired
+    private ProductMapper productMapper;
+
+    @PostMapping
+    public ResponseEntity<ProductAndCategoryDto> createProductAndCategoryDto(@RequestBody ProductAndCategoryDto productAndCategoryDto) {
```

Comment:
rename, you are saving an actual product with the help of a dto, not just creating an actual dto